src/components/post/index.js

```diff
@@ -43,13 +49,25 @@ const Post = ({
     openModal()
   }
 
-  const likeHandler = () => {
-    setIsLike(!isLike)
+  const likeHandler = async () => {
+    try {
+      await toggleLike(postId)
+      setUserLiked(!userLiked)
+      setLikesCount(userLiked ? likesCount - 1 : likesCount + 1)
+    } catch (error) {
+      console.error("Error toggling like:", error)
+    }
   }
 
-  const commentHandler = () => {
-    setIsComment(!isComment)
-  }
+  useEffect(() => {
+    const { userId } = jwtDecode(token)
```

Comment:
I think it would be good to add this functionality to the `auth` context. I can see a lot of different components using the user data, so having it available in the context would be a good idea. Then we could remove the `jwtDecode` import too.